### ISSUE TITLE
Documentation: Where to commit, what to change?

### ISSUE THREAD

**Issue by JensErat** [Homebrew/linuxbrew#292]
The documentation lacks some basic information for people wanting to contribute.

## Where to commit?

I'm currently having a look at the `task` formula. After seemingly failing to do a proper `brew update`/`git pull`, I made the same changes in parallel, and the resulting pull request broke in Travis.

Now I realized that's because a dependency on `libuuid` is missing for Linux. Adding a line

    depends_on "libuuid"

would suffice, but I also observed the pattern

    depends_on "libuuid" if OS.linux?

on the original (Mac) homebrew. Now: Where should I commit the change? It is Linux homebrew specific, so commit it here, or go to the Mac homebrew and add the second version there?

## What to modify anyway?

The Linux homebrew Formula contains mac-specific bottles. What to do about them, ignore them (as they don't hurt), or remove them?

## Is there any magic pulling involved?

Looking at the Formula's history, I can see very recent Mac-specific changes in the linuxbrew repository. Is there any magic involved pulling changes?

**Comment by sjackman** [comment]
> Where to commit?

Patches specific to Linux should be submitted to Homebrew/linuxbrew.

`depends_on "libuuid" unless OS.mac?` is the preferred idiom.

> What to modify anyway?

Ignore the Mac bottles

> Is there any magic pulling involved?

Homebrew/homebrew is merged into Homebrew/linuxbrew every week, usually on Monday.

**Comment by JensErat** [comment]
I created a new patch upstream, thank you for the explanation.

Is this already documented anywhere (and I've just been unable to find it)? Otherwise I'd propose some paragraph documenting this workflow, as this is a little bit unexpected to new contributors (or old ones from OS X homebrew).

**Comment by Mikaela** [comment]
> Is this already documented anywhere (and I've just been unable to find it)?

I haven't seen this anywhere, but I would like this to be documented in upstream CONTRIBUTING.md so it would be seen on both sides.

**Comment by sjackman** [comment]
Yes, I agree that CONTRIBUTING.md is missing some key information. I'll aim to update it this week.

**Comment by iggyvolz** [comment]
Should the apt-get information stay?  I'm working on testing all of the formula to see if they install on a clean Ubuntu VM, and from the looks of it many of them need new dependencies for linux-only.

Something worded like: We're in the process of making sure all dependencies are listed on the linux formula (if this is not the case, please open an issue if there is not one), but you may want to install the following programs to emulate a mac environment in the meantime: (current apt-get, yum, et cetera lines)?

**Comment by sjackman** [comment]
I'm going to leave the `apt-get` info for now. It'll make for a happier user not to run into frequent missing dependency errors. Please do open PR for missing dependencies as you find them.
